### DISCUSSION TITLE
Need Guidance to use the Publish_GitHub_Wiki_Content

### DISCUSSION THREAD

**Discussion by fslef** [gaelcolas/Sampler#426] [Q&A] [answered]
Hi Everyone,

I have started to create a new [PowerShell module](https://github.com/fslef/AzSecurityBuddy) using Sampler. After conducting several experiments, I believe I have a good understanding of the testing and building stages, even though I am still learning about Sampler.

I am now trying to publish to the GitHub wiki using the `Publish_GitHub_Wiki_Content task`, but I am running into an issue. Although I have configured my project for this task, I am unable to proceed further. The log indicates that the task is loaded, but the error message says that it is missing.

Here is the error message for your reference:

``` Powershell
[pre-build] Starting Build Init
[pre-build] Pre-pending '/Users/fslef/gitRepos/AzSecurityBuddy/output/RequiredModules' folder to PSModulePath
[build] Starting build with InvokeBuild.
[build] Parsing defined tasks
[build] Loading Configuration from /Users/fslef/gitRepos/AzSecurityBuddy/build.yaml
[build] Pre-pending '/Users/fslef/gitRepos/AzSecurityBuddy/output/module' folder to PSModulePath
Importing tasks from module Sampler
         Loading Build-Module.ModuleBuilder.build.Sampler.ib.tasks...
         Loading ChocolateyPackage.build.Sampler.ib.tasks...
         Loading Clean.ModuleBuilder.build.Sampler.ib.tasks...
         Loading Create_Changelog_Branch.build.Sampler.ib.tasks...
         Loading Create_Release_Git_Tag.build.Sampler.ib.tasks...
         Loading DeployAll.PSDeploy.build.Sampler.ib.tasks...
         Loading DscResource.Test.build.Sampler.ib.tasks...
         Loading generateHelp.PlatyPS.build.Sampler.ib.tasks...
         Loading GuestConfig.build.Sampler.ib.tasks...
         Loading Invoke-Pester.pester.build.Sampler.ib.tasks...
         Loading JaCoCo.coverage.build.Sampler.ib.tasks...
         Loading release.module.build.Sampler.ib.tasks...
         Loading Set-SamplerTaskVariable...
         Loading SetPsModulePath.build.Sampler.ib.tasks...
Importing tasks from module DscResource.DocGenerator
         Loading Task.Generate_Conceptual_Help...
         Loading Task.Generate_Wiki_Content...
         Loading Task.Publish_GitHub_Wiki_Content...
Importing tasks from module Sampler.GitHubTasks
         Loading New-Release.GitHub.build.Sampler.GitHubTasks.ib.tasks...
Adding Workflow from configuration:
  +-> .
  +-> build
  +-> publish
  +-> pack
  +-> test
[build] Executing requested workflow: publish
ERROR: Task 'publish': Missing task 'Publish_GitHub_Wiki_Content'.
At /Users/fslef/gitRepos/AzSecurityBuddy/build.ps1:327 char:13
+             task $workflow $workflowItem
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Build ABORTED /Users/fslef/gitRepos/AzSecurityBuddy/build.ps1. 0 tasks, 1 errors, 0 warnings 00:00:02.7272330
Invoke-Build.ps1: /Users/fslef/gitRepos/AzSecurityBuddy/build.ps1:507:9
Line |
 507 |          Invoke-Build @PSBoundParameters -Task $Tasks -File $MyInvocat …
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Task 'publish': Missing task 'Publish_GitHub_Wiki_Content'. At /Users/fslef/gitRepos/AzSecurityBuddy/build.ps1:327
     | char:13 +             task $workflow $workflowItem +    
```

Could you please suggest any ideas or troubleshooting tips to help me resolve this issue?

Thank you.
Francois

**Comment by fslef** [comment]
@gaelcolas @johlju by chance, do you have any ideas ?

**Reply by gaelcolas** [reply]
Yes, need to add the module to import the tasks.
Actually no, let me have a look when I'm not standing in a.train 😂

**Reply by raandree** [reply]
What @gaelcolas means is this: The task `Publish_GitHub_Wiki_Content` is not part of `Sampler` but of `Sampler.GitHubTasks`. You need to add this module to the `RequiredModules.psd1` and import the tasks provided by `Sampler.GitHubTasks` in the `build.yaml` like this:

```yml
ModuleBuildTasks:
  Sampler:
    - '*.build.Sampler.ib.tasks'
  Sampler.GitHubTasks:
    - '*.ib.tasks'
```

**Comment by fslef** [comment]
Thank you for your answer.

Except if I'm mistaken the task is loaded from the `DscResource.DocGenerator` module.
Both prerequisites seems to be met met:
- [The required module seems to be ok](https://github.com/fslef/AzSecurityBuddy/blob/main/RequiredModules.psd1#L16)
- [The module loading as well in the build.yaml](https://github.com/fslef/AzSecurityBuddy/blob/main/build.yaml#L144-L145)

And btw, the error message I copy and past in my initial message seems to indicate the task is well loaded:
```powershell
Importing tasks from module DscResource.DocGenerator
         Loading Task.Generate_Conceptual_Help...
         Loading Task.Generate_Wiki_Content...
         Loading Task.Publish_GitHub_Wiki_Content...
```

**Comment by johlju** [comment]
Have not seen this issue. Do you have a working branch somewhere that can be used as a repro?

**Comment by gaelcolas** [comment, marked as answer]
Ok, I have found your issue...
You have the following:

```yaml
  DscResource.DocGenerator:
    - 'Tasks.*'
```

While it should be:

```yaml
  DscResource.DocGenerator:
    - 'Task.*'
```

Yup, just an extra s... :)

**Comment by fslef** [comment]
Thank you, Gael.

You're right that it resolved the issue.

However, the initial error message is still unusual because it suggested that the task was loaded correctly, which bring the confusion.
